File: Data Structures & Algorithms/find-pivot-index/submission-0.java
class Solution {
    public int pivotIndex(int[] nums) {
        int n=nums.length;
        int[] ls= new int[n];
        ls[0] = nums[0];
        int[] rs=new int[n];
        rs[n-1]= nums[n-1];
        for (int i=1;i<n;i++){
            ls[i]=ls[i-1]+nums[i];
        }
        for (int i=n-2;i>=0;i--){
            rs[i]=rs[i+1]+nums[i];
        }
        for (int i=0;i<n;i++){
            if (ls[i]-nums[i]==rs[i]-nums[i]){
                return i;
            }
        }
        return -1;
        
    }
}

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-0.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        left=0
        min_length = float('inf')
        curr=0
        for right in range(len(nums)):
            curr+=nums[right]

            while curr>=target:
                curr-=nums[left]
                min_length=min(min_length,right-left+1)
                left+=1
                
        if min_length==float('inf'):
            return 0
        return min_length
        